### PR TITLE
Add Modulus Operator

### PR DESCRIPTION
**Pull request by redwizard42**
Adds the modulus operator to allow scaling to the remainder of a division. Most useful for anytime you have an increasing index that in practice has looping behavior without its value actually looping.

Example: A Game with 5 stages that loop, but the 0-based stage identified only ever increases, with no loop counter is present in RAM. With modulus operator, the developer could check that you are on 'any third stage' in the sequence using "Add Source Stage % 5 = Value 2"

Currently only modulo using a power of two can be accomplished (with Bitwise And)

### CODE REVIEW

File: test/rcheevos/test_condition.c [hunk 1]
```diff
@@ -372,6 +372,7 @@ void test_condition(void) {
   TEST_PARAMS5(test_parse_condition, "A:0xH1234/8", RC_CONDITION_ADD_SOURCE, RC_OPERAND_ADDRESS, RC_OPERATOR_DIV, 0);
   TEST_PARAMS5(test_parse_condition, "A:0xH1234&8", RC_CONDITION_ADD_SOURCE, RC_OPERAND_ADDRESS, RC_OPERATOR_AND, 0);
   TEST_PARAMS5(test_parse_condition, "A:0xH1234^8", RC_CONDITION_ADD_SOURCE, RC_OPERAND_ADDRESS, RC_OPERATOR_XOR, 0);
+  TEST_PARAMS5(test_parse_condition, "A:0xH1234%8", RC_CONDITION_ADD_SOURCE, RC_OPERAND_ADDRESS, RC_OPERATOR_MOD, 0);
```
Comment (Jamiras) [inline test/rcheevos/test_condition.c:375]: Please also add some tests for processing the logic. Here's some examples for other operators in test_value:
https://github.com/RetroAchievements/rcheevos/blob/3decda081f98582e02d583474048dade799edb2a/test/rcheevos/test_value.c#L534-L539

File: src/rcheevos/rc_validate.c [hunk 2]
```diff
@@ -143,6 +143,11 @@ static uint32_t rc_scale_value(uint32_t value, uint8_t oper, const rc_operand_t*
     case RC_OPERATOR_XOR:
       return value | rc_max_value(operand);
 
+    case RC_OPERATOR_MOD:
+    {
+      return value;
+    }
```
[the remark this answers is not in the document]
Comment (redwizard42) [reply src/rcheevos/rc_validate.c:150]: Corrected. Had something equivalent and overthought it before.

File: test/rcheevos/test_condition.c (hunk 1 above)
Comment (redwizard42) [reply test/rcheevos/test_condition.c:375]: Added a handful.  Missing something needed to run the tests. Will need to sort that out.

Comment (redwizard42) [reply test/rcheevos/test_condition.c:375]: Just didn't realize I run the tests buy running in debugger.  Corrected a few. Added some with floats. Should be ready. Let me know if there's anything else I ought to test.

**Comment by redwizard42** [comment]
I corrected the extra comma in the operator enum. Habit from C# development where I do it all the time.

File: test/rcheevos/test_condition.c (hunk 1 above)
Comment (redwizard42) [reply test/rcheevos/test_condition.c:375]: Added typed value tests for modulus.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**